md4c/md4c.c

```diff
@@ -3630,6 +3649,26 @@ md_analyze_tilde(MD_CTX* ctx, int mark_index)
     }
 }
 
+static void
+md_analyze_dollar(MD_CTX* ctx, int mark_index)
+{
+    /* This should mimic the way inline equations work in LaTeX, so there
+     * can only ever be one item in the chain (i.e. the dollars can't be
+     * nested). This is basically the same as the md_analyze_tilde function.
+     *
+     * Note: Doing things this way means we are very lenient and accept and
+     * number (1,2) of $ to close an equation. */
+    if(DOLLAR_OPENERS.head >= 0) {
+        /* The chain already contains an opener, so we are the closer. */
+        int opener_index = DOLLAR_OPENERS.head;
+
+        md_rollback(ctx, opener_index, mark_index, MD_ROLLBACK_CROSSING);
+        md_resolve_range(ctx, &DOLLAR_OPENERS, opener_index, mark_index);
+    } else {
+        md_mark_chain_append(ctx, &DOLLAR_OPENERS, mark_index);
+    }
+}
```

Comment:
Imho, it should not allow mixing `$` and `$$` (or vice versa). When that happens, I would simply discard the old opener, and use the new one as a potential opener instead.

That would mean that e.g.
```
$foo$$bar$$
```
sees the `bar` as a (display) equation and not `foo`. It's imho more intuitive behavior.

Reply:
Yes, I think that would make more sense. It can lead to confusing situations where $blub$$ ... lots of text $$some display eqn$$ would be parsed in a wired way, but I think that's overall better than the alternative (which can also lead to wired behaviour.)